### ISSUE TITLE
Green dot disconnected automatically after a period of time

### ISSUE THREAD

**Issue by wx-Yao** [allinurl/goaccess#1491]
Hi :)

Thanks for the great project. I've got everything running and everything's cool. The goaccess is hosted on my VPS and I reverse proxy the goaccess and ws using Nginx in realtime http mode. I read the log via my Mac browser.

Now the problem is, if I open the web interface, everything works. And there's a green dot below to gear/bar icon (I guess that indicates the websocket is connected successfully). But after like 1 mins or 2, the greed dot turns gray and the HTML isn't updated any more.

The websocket on the server side is still working, I can confirm this (by netstat). And the console output in my browser is empty, no error.

Is this app designed to work like this? or is there anything I was missing？

Tell me if you need any addtional info. Thank you so much. :)

**Comment by allinurl** [comment]
Are you running the latest version?

It's not designed that way. Take a look at the https://rt.goaccess.io
I'll need more info on this as I can't really help without the details. Can you please run goaccess with `strace` and see what is it doing when it hangs, please post the output here. Thanks.

**Comment by wx-Yao** [comment]
I'm running the latest version. The `--version` gives:

```
GoAccess - 1.3.
For more details visit: http://goaccess.io
Copyright (C) 2009-2016 by Gerardo Orellana

Build configure arguments:
  --enable-utf8
  --enable-geoip=mmdb
  --enable-tcb=memhash
  --with-getline
  --with-openssl
```
which includes my compling options.

I ran the strace and record until the green dot grayed out. Then i hit Ctrl-c. Here's strace output ( strace -o strace.txt bla). Hopefully I didn't leak anything sensitive though it's personal server. Those log doesn't make a lot of sense to me lol.

[strace.txt](https://github.com/allinurl/goaccess/files/3355266/strace.txt)

P.S. I've read many issues and it seems you were helping many people debugging lol. Thank you so much.

**Comment by wx-Yao** [comment]
Any idea what happened pls :)

**Comment by allinurl** [comment]
I don't see any issues other than you hitting a `SIGINT`. Does it always happen at the same time or it can vary? Are you able to try this on a different machine? Can you run it with the reverse proxy as it could be the issue as well? Please post the results here. Thanks.

**Comment by allinurl** [comment]
** without the reverse proxy

**Comment by wx-Yao** [comment]
@allinurl Hi sorry for the late reply. I was onto something else for the past weeks. sorry about that

And, the good news is I got the issue resolved which is kinda important for those who wanna proxy the goaccess using Nginx, especially whose sites are not popular enough to get updated data fed to the client regularly.

So the cause of this problem is quite simple. I proxied the Goaccess web server as well as the websocket behind the Nginx. And it's because my site isn't popular enough, the nginx access log doesn't grow for some time, which cannot feed the websocket with new data, and the connection just timeout. The normal death of the connection doesn't have any error log.

**Remediation**: in the Nginx config, set the `proxy_read_timeout` to big number (10 min for example). It did keep my green dot on. But Nginx official doc also suggested the proxied server, which is goaccess, to send ping packet to the connection. I don't know if the tool did this. For now I just extended the `proxy_read_timeout`. It works fine.

I'll keep monitoring to see if the green dot goes off.
